packages/components/src/internal/util/utils.ts

```diff
@@ -378,12 +378,22 @@ export function getUpdatedDataFromGrid(
         const originalRow = originalGridData.get(id.toString());
         if (originalRow) {
             const row = editedRow.reduce((row, value, key) => {
+                // We can skip the idField for the diff check, that will be added to the updated rows later
+                if (key === idField) return row;
+
                 let originalValue = originalRow.has(key) ? originalRow.get(key) : undefined;
                 const col = queryInfo.getColumn(key);
                 const isDate = col?.jsonType === 'date';
                 // Convert empty cell to null
                 if (value === '') value = null;
 
+                // Lookup columns store a list but grid only holds a single value
+                if (List.isList(originalValue) && !Array.isArray(value)) {
+                    originalValue = Map.isMap(originalValue.get(0))
+                        ? originalValue.get(0).get('value')
+                        : originalValue.get(0).value;
+                }
```

Comment:
Yes, I believe moving this if statement up, is now pulling out the originalValue from the list so that it will now go through the else if statement below that DOES do the date handling (i.e. starting at line 421). That is, we no longer need that call to `(isDate ? getJsonDate...` in two places.